main/utils/artifactory.py

```diff
@@ -28,8 +28,23 @@ def receive_build_info(self, release_request):
 
     def promote(self, release_request, buildinfo, revoke=False):
         status = 'released'
+        try:
+            sourcerepo, targetrepo = buildinfo.get_source_and_target_repos(revoke)
+            print(f"Promoting build {release_request.project}#{release_request.buildnumber} from {sourcerepo} to "
+                  f"{targetrepo}")
 
-        if buildinfo.is_multi():
+            url = f"{self.url}/api/build/promote/{release_request.project}/{release_request.buildnumber}"
+            if revoke:
+                status = "it-passed"
+            json_payload = {
+                "status": f"{status}",
+                "sourceRepo": f"{sourcerepo}",
+                "targetRepo": f"{targetrepo}"
+            }
+            r = requests.post(url, data=json.dumps(json_payload), headers=self.headers)
+        except KeyError:
+            # The build promotion used the homemade user plugin multipromote
+            # In this case, the release status does not have the key 'repository' set
```

Comment:
Could explain a bit more that we're going to get the repository from another source?